### PR TITLE
Sync pokemon expansion

### PR DESCRIPTION
**Pull request by ExpoSeed**

### CODE REVIEW

**Review by AsparagusEduardo — DISMISSED** [review]

**Review by ultima-soul — CHANGES_REQUESTED** [review]
In addition to changes requested above, please change POKEMON_SLOTS_NUMBER from 808 to 810 since the number wasn't incremented when Meltan and Melmetal were added.

**Comment by ExpoSeed** [comment]
All changes addressed in latest commit.

**Review by ultima-soul — APPROVED** [review]

**Review by AsparagusEduardo — DISMISSED** [review]

**Review by ultima-soul — DISMISSED** [review]

File: include/global.h [hunk 1]
```diff
@@ -115,6 +114,7 @@
 
 #define ROUND_BITS_TO_BYTES(numBits)(((numBits) / 8) + (((numBits) % 8) ? 1 : 0))
 
+#define POKEMON_SLOTS_NUMBER 810
```
Comment (ghoulslash) [inline include/global.h:117]: I think this should be defined in constants/species.h and/or changed to reflect the NUM_SPECIES change from pret

Comment (ExpoSeed) [reply include/global.h:117]: It was originally defined right below the floating point stuff, and placing it here avoids a merge conflict in BE-PE merges.

Comment (ExpoSeed) [reply include/global.h:117]: Also in PE, `NUM_SPECIES != POKEMON_SLOTS_NUMBER` because it is used differently

**Review by AsparagusEduardo — APPROVED** [review]

**Pull request merged**